### PR TITLE
Update and delete value in the table

### PR DESCRIPTION
**Pull request by Iptysam**
- Add a column species of type string to your animals table. Modify your schema.sql file.
- Insert the following data:
    - Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
    - Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
    - Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
    - Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
    - Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
    - Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
    - Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.
- Remember to add these insertions to your data.sql file.
- Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
- Inside a transaction:
    - Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
    - Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
    - Commit the transaction.
    - Verify that change was made and persists after commit.
    - Take a screenshot of the results of your actions.
- Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.
- After the rollback verify if all records in the animals table still exists. After that, you can start breathing as usual ;)
- Take a screenshot of the results of your actions.
- Inside a transaction:
    - Delete all animals born after Jan 1st, 2022.
    - Create a savepoint for the transaction.
    - Update all animals' weight to be their weight multiplied by -1.
    - Rollback to the savepoint
    - Update all animals' weights that are negative to be their weight multiplied by -1.
    - Commit transaction
    - Take a screenshot of the results of your actions.
    - Remember to add all these transactions to your queries.sql file.
- Write queries to answer the following questions:
    - How many animals are there?
    - How many animals have never tried to escape?
    - What is the average weight of animals?
    - Who escapes the most, neutered or not neutered animals?
    - What is the minimum and maximum weight of each type of animal?
    - What is the average number of escape attempts per animal type of those born between 1990 and 2000?
    - Take a screenshot of the results of your queries.
- 
- count where escape 0
- count animals
- 
- Results of the transactions
UPDATE NEG VALUE
delete after 2022
update mon names

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -9,7 +9,8 @@ CREATE TABLE IF NOT EXISTS public.animals
     date_of_birth integer,
     escape_attempts integer,
     neutered boolean,
-    weight_kg numeric
+    weight_kg numeric,
+    species text COLLATE pg_catalog."default"
```
Comment (mikethreels) [inline schema.sql:13]: - [ ] to add a column to an existing table we need to use the `ALTER_TABLE` query.

File: queries.sql [hunk 2]
```diff
@@ -9,3 +9,64 @@ SELECT * FROM animals WHERE name NOT IN ('Gabumon');
...
 
+
+/*Transaction*/
+UPDATE animals 
+SET species = 'unspecified';
+
+SELECT species from animals;
+ROLLBACK;
+
+SELECT species from animals;
```
Comment (mikethreels) [inline queries.sql:20]: - [ ] you need to add `BEGIN` in front of the query to start it.

File: queries.sql [hunk 3]
```diff
@@ -9,3 +9,64 @@ SELECT * FROM animals WHERE name NOT IN ('Gabumon');
...
+
+/*Transaction2*/
+UPDATE animals SET species = 'dogomun' WHERE name like '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+
+SELECT species from animals;
+COMMIT;
+SELECT species from animals;
+
```
Comment (mikethreels) [inline queries.sql:29]: - [ ] you need to add `BEGIN` in front of the query to start it.

File: queries.sql [hunk 4]
```diff
@@ -9,3 +9,64 @@ SELECT * FROM animals WHERE name NOT IN ('Gabumon');
...
+SAVEPOINT S1;
+
+DELETE FROM animals WHERE date_of_birth > '20220101';
+
+SAVEPOINT 2;
+
+UPDATE animals SET weight_kg = weight_kg * -1;
+
+ROLLBACK TO S2;
```
Comment (mikethreels) [inline queries.sql:51]: - [ ] the savepoint is called 2 and you are rolling back to savepoint S2

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Ipty, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — APPROVED** [review]
# Hi @Iptysam, 👋🏻👋🏻 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉🚀🔥🎉🎉 


## Highlights ✨✨
- Followed Gitflow ✅
- Great working on the functionality 🔥🚀🎉
- Professional README file 💯✨🔥
- Descriptive PR ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**